Sources/Subs-Db-mysql.php

```diff
@@ -100,7 +100,13 @@ function smf_db_initiate($db_server, $db_name, $db_user, $db_passwd, $db_prefix,
 	if (empty($db_options['dont_select_db']) && !@mysqli_select_db($connection, $db_name) && empty($db_options['non_fatal']))
 		display_db_error();
 
-	mysqli_query($connection, 'SET SESSION sql_mode = \'ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION\'');
+	$sql_mode = '';
+	// Set the mysql default mode
+	$sql_mode .= (empyt($sql_mode) ? '' : ',') . 'ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';
+	// Disable backlash escape, only '' is than a valid one
+	$sql_mode .= (empyt($sql_mode) ? '' : ',') . 'NO_BACKSLASH_ESCAPES';
+
+	mysqli_query($connection, "SET SESSION sql_mode = '" . $sql_mode . "'");
```

Comment:
There are typos in the lines above. Since those need to be fixed anyway, why not just put everything back on a single line?
```
mysqli_query($connection, "SET SESSION sql_mode = 'ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION,NO_BACKSLASH_ESCAPES'");
```

Reply:
i was guessing,
that i would look me cleaner,
when the different set of mods are grouped and mention in the comment what they do and
when you want to disable a specif behavior that you only comment out the right line.

Reply:
Well, if you want to do that, the cleanest way would be to use an array and then implode it to make the final string. Something like this, perhaps:
```
$sql_mode = array(
	'ONLY_FULL_GROUP_BY',
	'STRICT_TRANS_TABLES',
	'NO_ZERO_IN_DATE',
	'NO_ZERO_DATE',
	'ERROR_FOR_DIVISION_BY_ZERO',
	'NO_AUTO_CREATE_USER',
	'NO_ENGINE_SUBSTITUTION',
	'NO_BACKSLASH_ESCAPES',
);
mysqli_query($connection, "SET SESSION sql_mode = '" . implode(',', $sql_mode) . "'");
```

If you want to make it really friendly for mods, you could insert an integration hook between the array and the `mysqli_query()` call in order to give the mods edit-free access to the `$sql_mode` array.

Reply:
Mods we can't support here,
Since the connection to the db is not up